runtime/src/main/java/org/corfudb/runtime/clients/NettyClientRouter.java

```diff
@@ -330,6 +352,12 @@ private void addReconnectionOnCloseFuture(@Nonnull Channel channel,
             // If we aren't shutdown, reconnect.
             if (!shutdown) {
                 log.info("addReconnectionOnCloseFuture[{}]: reconnecting", node);
+                if (sslAuthenticationFailure.get()) {
+                    retryPolicy.nextWait();
+                } else {
+                    // reset exponential time retry policy, to start from beginning on next failure.
+                    retryPolicy.reset();
```

Comment:
This seems a bit strange if you have an object that encapsulates the exponential backoff logic why do you need to invoke reset externally?  Shouldn't retryPolicy.nextWait reset itself internally ?

Reply:
This reset happens in between authentication failures and successes to reset the timer calculation from the beginning and exponentially backoff from the start. Otherwise, in an scenario failure, success, failure, this last would never restart it would continue from the rate of the last backoff.